python/lsst/analysis/drp/colorColorFitPlot.py

```diff
@@ -75,6 +77,72 @@ class ColorColorFitPlotConfig(pipeBase.PipelineTaskConfig,
         default={"xMin": 0.1, "xMax": 0.2, "yMin": 0.1, "yMax": 0.2, "mHW": 0.5, "bHW": 0.0}
     )
 
+    bands = pexConfig.DictField(
+        doc="Names of the bands to use for the colors.  Plots are of band2 - band3 vs. band2 - band1",
+        keytype=str,
+        itemtype=str,
+        default={"band1": "g", "band2": "r", "band3": "i"}
+    )
+
+    fluxTypeForColor = pexConfig.Field(
+        doc="Flavor of flux measurement to use for colors",
+        default="psfFlux",
+        dtype=str,
+    )
+
+    xLims = pexConfig.ListField(
+        doc="Minimum and maximum x-axis limit to force (provided as a list of [xMin, xMax]). "
+        "If `None`, limits will be computed and set based on the data.",
+        dtype=float,
+        default=None,
+        optional=True,
+    )
+
+    yLims = pexConfig.ListField(
+        doc="Minimum and maximum y-axis limit to force (provided as a list of [yMin, yMax]). "
+        "If `None`, limits will be computed and set based on the data.",
+        dtype=float,
+        default=None,
+        optional=True,
+    )
+
+    def setDefaults(self):
+        super().setDefaults()
+        self.axisActions.xAction.magDiff.returnMillimags = False
+        self.axisActions.yAction.magDiff.returnMillimags = False
+        self.axisActions.magAction.column = "r_psfFlux"
+        self.stellarLocusFitDict = {"xMin": 0.28, "xMax": 1.0, "yMin": 0.02, "yMax": 0.48,
+                                    "mHW": 0.52, "bHW": -0.08}
+        self.xLims = (-0.7, 2.3)
+        self.yLims = (-0.7, 2.6)
+        self.setConfigDependencies()
+
+    def setConfigDependencies(self):
+        # The following config settings are conditional on other configs.
+        # Set them based on the inter-dependencies here to ensure they are
+        # in sync.  This can (and should) be called in the pipeline definition
+        # if any of the inter-dependent configs are chaned (e.g. self.bands
+        # or self.fluxTypeForColor here.  See plot_wFit_CModel in
+        # stellarLocusPlots.yaml for an example use case.
```

Comment:
I like this way of setting inter-dependent configs, since I've wondered something similar for rho statistics. I have enforced that they are satisfied in a `validate` method, and I think there should be a `validate` method here as well, to make sure that the config doesn't become invalid by doing something incorrectly after calling. `setConfigDependencies`

Reply:
Heh...it felt totally ugly when I was writing it, but I couldn't think of a better way (I seem to recall slinking into Jim's office asking just how bad this idea was...yes, ugly, but not forbidden!)

Reply:
Just to make sure I understand, are you suggesting this `validate` should happen in `run` (i.e. this is different than the `Config` class `validate`?  I'm pretty sure I tried to make that work, but could never be sure it executed after all possible config manipulations...)?

Reply:
No, I'm just saying you should have a config `validate` that checks the invariants instead of relying only on this method having been called to set up the config.  It's the job of whatever is running the task to call the config `validate` and then freeze the config.

Reply:
Ah, ok.  I think I am again going to request that this be added in the ticket to update `analysis_tools`, since I would prefer not to have to do all the necessary "validation" (pun intended) of the `validate` function.  Please push back if you don't think that's a reasonable request and want it added on this ticket!

Reply:
Definitely; the configuration will certainly have to change somewhat for that transition anyway.